pom.xml

```diff
@@ -74,7 +74,10 @@
         <dep.docker-java.version>3.3.0</dep.docker-java.version>
         <dep.jayway.version>2.6.0</dep.jayway.version>
         <dep.ratis.version>2.2.0</dep.ratis.version>
+        <dep.errorprone.version>2.18.0</dep.errorprone.version>
+        <dep.guava.version>32.1.0-jre</dep.guava.version>
         <dep.jackson.version>2.11.0</dep.jackson.version>
+        <dep.j2objc.version>2.8</dep.j2objc.version>
```

Comment:
Do we need j2bjc? I see it in a few poms, but not in Java code anywhere. Can we instead remove it where it appears?

Reply:
Yes we need it. It is often used in the guava code, e.g. in `com.google.common.util.concurrent.AbstractFuture` class, and this class is used via `com.google.common.util.concurrent.SettableFuture` by `com.facebook.presto.hive.HiveSplitSource`

Reply:
but shouldn't guava supply that then? Does it need to be in our poms? 

Also should we just use the guava BOM instead of setting these dependencies up manually?

Reply:
We are pinning the version to avoid conflicts among different versions of the same dependency (Maven Enforcer plugin).